python-mini-projects/projects/Web_page_summation/utils/comparison.py

```diff
@@ -0,0 +1,50 @@
+
+# https://github.com/chakki-works/sumeval
+# https://github.com/Tian312/awesome-text-summarization
+
+from sumeval.metrics.rouge import RougeCalculator
+from sumeval.metrics.bleu import BLEUCalculator
+
+
+def eval_rouges(refrence_summary, model_summary):
+    # refrence_summary = "tokyo shares close up #.## percent"
+    # model_summary = "tokyo stocks close up # percent to fresh record high"
+
+    rouge = RougeCalculator(stopwords=True, lang="en")
+
+    rouge_1 = rouge.rouge_n(
+        summary=model_summary,
+        references=refrence_summary,
+        n=1)
+
+    rouge_2 = rouge.rouge_n(
+        summary=model_summary,
+        references=[refrence_summary],
+        n=2)
+
+    rouge_l = rouge.rouge_l(
+        summary=model_summary,
+        references=[refrence_summary])
+
+    # You need spaCy to calculate ROUGE-BE
+
+    rouge_be = rouge.rouge_be(
+        summary=model_summary,
+        references=[refrence_summary])
+
+    bleu = BLEUCalculator()
+    bleu_score = bleu.bleu(summary=model_summary,
+                           references=[refrence_summary])
+
+    # print("ROUGE-1: {}, ROUGE-2: {}, ROUGE-L: {}, ROUGE-BE: {}".format(
+    #    rouge_1, rouge_2, rouge_l, rouge_be
+    # ).replace(", ", "\n"))
+
+    return rouge_1, rouge_2, rouge_l, rouge_be, bleu_score
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears you are unpacking more than three variables while using a function that returns more than one variable. Using a large number of return values is prone to errors. We recommend that you return a small class or `namedtuple` instance instead.

[Learn more](https://docs.python.org/3/library/typing.html#typing.NamedTuple)